File: c/leap/leap.c
#include "leap.h"

#define LEAP_YEAR     true
#define NOT_LEAP_YEAR false

bool leap_year(int testYear)
{
	bool isLeap = LEAP_YEAR;
	
	if(testYear % 4 == 0)
	{
		if(testYear % 100 == 0)
		{
			if(testYear % 400 != 0)
			{
				isLeap = NOT_LEAP_YEAR;
			}
		}
    }
	else
	{
		isLeap = NOT_LEAP_YEAR;
	}

	return isLeap;
}

File: python/leap/leap.py
"""
FUNCTION: is_leap_year
ARGS:    int (year)
RETURNS: bool
DESCRIPTION: TRUE - if input_year is a leap year
             FALSE - if input_year is not a leap year
"""

def leap_year(year):
    return (year % 4 == 0) and (year % 100 != 0 or year % 400 == 0)